### PR TITLE
tinygltf: inline in draco

### PR DESCRIPTION
**Pull request by jansol**
As discussed in #228937 this is a header-only library whose only user is draco. Inline it in the draco package since there are no other users and upgrading it to 2.8.8 breaks the draco build.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/tinygltf/default.nix [hunk 1]
```diff
@@ -1,30 +0,0 @@
-{ lib
```
Comment (AndersonTorres) [inline pkgs/development/libraries/tinygltf/default.nix:1]: Don't remove this file. Move it to `draco/tinygltf.nix`.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -13201,7 +13201,10 @@ with pkgs;
 
   tinyfecvpn = callPackage ../tools/networking/tinyfecvpn { };
 
-  tinygltf = callPackage ../development/libraries/tinygltf { };
+  tinygltf = throw ''
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:13204]: Put this in `top-level/aliases.nix` .

File: pkgs/development/libraries/draco/default.nix [hunk 3]
```diff
@@ -9,20 +9,31 @@
 , withTranscoder ? true
 , eigen
 , ghc_filesystem
-, tinygltf
 }:
 
 let
   cmakeBool = b: if b then "ON" else "OFF";
+  tinygltf = stdenv.mkDerivation (finalAttrs: {
```
Comment (AndersonTorres) [inline pkgs/development/libraries/draco/default.nix:16]: After this you can simply write in all-packages.nix:

```nix
draco =  callPackage ../development/libraries/draco {
   tinygltf = callPackage ../development/libraries/draco/tinygltf.nix {};
. . . 
};
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (Janik-Haag) [reply pkgs/top-level/all-packages.nix:13204]: Isn't this still open, @AndersonTorres what changed your mind?

Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:13204]: Nothing, it is just there because someone forgot to hit the button.

**Pull request merged**